libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/transition/MapboxNavigationCameraTransition.kt

```diff
@@ -31,103 +29,14 @@ class MapboxNavigationCameraTransition(
-                ).toLong()
-                .coerceAtMost(MAXIMUM_LOW_TO_HIGH_DURATION)
-            val centerAnimator = cameraPlugin.createCenterAnimator(
-                CameraAnimatorOptions.cameraAnimatorOptions(center) {
-                    owner(NAVIGATION_CAMERA_OWNER)
-                }
-            ) {
-                duration = centerDuration
-                interpolator = SLOW_OUT_SLOW_IN_INTERPOLATOR
-            }
-            animators.add(centerAnimator)
-        }
-
-        var zoomDelay = 0.0
-        var zoomDuration = 0L
-        cameraOptions.zoom?.let { zoom ->
-            val currentMapCameraZoom = currentMapCameraState.zoom
-            val zoomDelta = abs(zoom - currentMapCameraZoom)
-            val zoomAnimationRate = 2.2
-            zoomDelay = centerDuration * 0.5
-            zoomDuration = ((zoomDelta / zoomAnimationRate) * 1000.0).toLong()
-                .coerceAtMost(MAXIMUM_LOW_TO_HIGH_DURATION)
-
-            val zoomAnimator = cameraPlugin.createZoomAnimator(
-                CameraAnimatorOptions.cameraAnimatorOptions(zoom) {
-                    owner(NAVIGATION_CAMERA_OWNER)
-                }
-            ) {
-                startDelay = zoomDelay.toLong()
-                duration = zoomDuration
-                interpolator = SLOW_OUT_SLOW_IN_INTERPOLATOR
-            }
-            animators.add(zoomAnimator)
-        }
-
-        cameraOptions.bearing?.let { bearing ->
-            val bearingShortestRotation = normalizeBearing(currentMapCameraState.bearing, bearing)
-            val bearingDuration = 1800.0
-            val bearingDelay = (zoomDelay + zoomDuration - bearingDuration).coerceAtLeast(0.0)
-            val bearingAnimator = cameraPlugin.createBearingAnimator(
-                CameraAnimatorOptions.cameraAnimatorOptions(bearingShortestRotation) {
-                    owner(NAVIGATION_CAMERA_OWNER)
-                }
-            ) {
-                startDelay = bearingDelay.toLong()
-                duration = bearingDuration.toLong()
-                interpolator = SLOW_OUT_SLOW_IN_INTERPOLATOR
-            }
-            animators.add(bearingAnimator)
-        }
-
-        val pitchAndPaddingDuration = 1200.0
-        val pitchAndPaddingDelay = (
-            zoomDelay + zoomDuration - pitchAndPaddingDuration + 100
-            )
-            .coerceAtLeast(0.0)
-        cameraOptions.pitch?.let { pitch ->
-            val pitchAnimator = cameraPlugin.createPitchAnimator(
-                CameraAnimatorOptions.cameraAnimatorOptions(pitch) {
-                    owner(NAVIGATION_CAMERA_OWNER)
-                }
-            ) {
-                startDelay = pitchAndPaddingDelay.toLong()
-                duration = pitchAndPaddingDuration.toLong()
-                interpolator = SLOW_OUT_SLOW_IN_INTERPOLATOR
-            }
-            animators.add(pitchAnimator)
-        }
+        val pluginImpl = cameraPlugin as CameraAnimationsPluginImpl
```

Comment:
Did we double-check that developers cannot provide their own implementations of the `CameraAnimationsPlugin`? If they can, we should safe-cast and fallback to the old solution.